cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -122,6 +132,35 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     String subnet = conf.getSubnet();
     Network networkInfo = getNetworkInfo(projectId, network, compute);
 
+    List<NetworkPeering> peerings = networkInfo.getPeerings();
+    for (NetworkPeering peering : peerings) {
+      Matcher matcher = GCP_NETWORK_FULL_PATH.matcher(peering.getNetwork());
+      boolean match = matcher.find();
+      if (match) {
+        String peeredProjectID = matcher.group(2);
+        String peeredNetwork = matcher.group(4);
+        if (peeredProjectID.equals(systemProjectId) && peeredNetwork.equals(systemNetwork)) {
+          if (peering.getState().equals("ACTIVE")) {
+            // peering to tenant project found prefer internal unless prefer external IP is set
```

Comment:
remove comment about "tenant project"

Reply:
also remove it from the error message and the PR title?